frontends/web/tests/aopp.test.ts

```diff
@@ -0,0 +1,148 @@
+let simulatorProc : ChildProcess | undefined;
+
+test('AOPP', async ({ page, host, frontendPort, servewalletPort }, testInfo) => {
+
+
+  await test.step('Start regtest and init wallet', async () => {
+    regtest = await launchRegtest();
+    // Give regtest some time to start
+    await new Promise((resolve) => setTimeout(resolve, 3000));
+    await setupRegtestWallet();
+  });
+
+
+  await test.step('Start servewallet', async () => {
+    servewallet = new ServeWallet(page, servewalletPort, frontendPort, host, testInfo.title, testInfo.project.name, { regtest: true, testnet: false, simulator: true });
+    await servewallet.start();
+  });
+
+  await test.step('Start simulator', async () => {
+    const simulatorPath = process.env.SIMULATOR_PATH;
+    if (!simulatorPath) {
+      throw new Error('SIMULATOR_PATH environment variable not set');
+    }
+
+    simulatorProc = startSimulator(simulatorPath, testInfo.title, testInfo.project.name, true);
+    console.log('Simulator started');
+  });
+
+
+  await test.step('Initialize wallet', async () => {
+    await completeWalletSetupFlow(page);
+  });
+
+  let recvAdd: string;
+  await test.step('Grab receive address', async () => {
+    await page.getByRole('link', { name: 'Bitcoin Regtest Bitcoin' }).click();
+    await page.getByRole('button', { name: 'Receive RBTC' }).click();
+    await page.getByRole('button', { name: 'Verify address on BitBox' }).click();
+    const addressLocator = page.locator('[data-testid="receive-address"]');
+    recvAdd = await addressLocator.inputValue();
+    console.log(`Receive address: ${recvAdd}`);
+  });
+
+  await test.step('Send RBTC to receive address', async () => {
+    await page.waitForTimeout(2000);
+    const sendAmount = '10';
+    await sendCoins(recvAdd, sendAmount);
+    await mineBlocks(12);
+    console.log(`Sent ${sendAmount} RBTC to ${recvAdd}`);
+  });
+
+
+  let aoppRequest: string;
+  await test.step('Start AOPP server and generate AOPP request', async () => {
+    console.log('Starting AOPP server...');
+    aoppServer = await startAOPPServer();
+    console.log('AOPP server started.');
+    console.log('Generating AOPP request...');
+    aoppRequest = await generateAOPPRequest('rbtc');
+    console.log(`AOPP Request URI: ${aoppRequest}`);
+  });
+
+  await test.step('Kill servewallet and restart with AOPP request', async () => {
+    await servewallet.stop();
+    console.log('Servewallet stopped.');
+    servewallet = new ServeWallet(page, servewalletPort, frontendPort, host, testInfo.title, testInfo.project.name, { regtest: true, testnet: false, simulator: true });
+    await servewallet.start({ extraFlags: { aoppUrl: aoppRequest } });
+    console.log('Servewallet restarted with AOPP request.');
+  });
+
+  let aoppAddress : string | null;
+  await test.step('Verify AOPP flow is in progress', async () => {
+    await page.goto('/');
+    await page.getByRole('button', { name: 'Continue' }).click();
```

Comment:
Before clicking the button, could you test that the app shows "\<HOST\> is requesting a receiving address."?